log/logr/packetlogger.go

```diff
@@ -155,3 +155,48 @@ func errLogsToStderr(c zap.Config) zap.Option {
 	})
 	return splitLogger
 }
+
+// handleFields converts a bunch of arbitrary key-value pairs into Zap fields.  It takes
+// additional pre-converted Zap fields, for use with automatically attached fields, like
+// `error`.
+func handleFields(l *zap.Logger, args []interface{}, additional ...zap.Field) []zap.Field {
+	// a slightly modified version of zap.SugaredLogger.sweetenFields
+	if len(args) == 0 {
+		// fast-return if we have no suggared fields.
+		return additional
+	}
+
+	// unlike Zap, we can be pretty sure users aren't passing structured
+	// fields (since logr has no concept of that), so guess that we need a
+	// little less space.
+	fields := make([]zap.Field, 0, len(args)/2+len(additional))
+	for i := 0; i < len(args); {
```

Comment:
you can do the `i += 2` here instead of at the end of the loop body.

Reply:
fyi, this is a copy/paste from [here](https://github.com/go-logr/zapr/blob/146009e52d528183a25bf1a1e3cf56d1ff3919b5/zapr.go#L79),

Reply:
oic, then it makes sense to keep it as is. Please add some info to the doc comment that this came from that url.